### PR TITLE
Release 2.4.0

### PR DESCRIPTION
**Pull request by algesten**
PR for releasing 2.4.0.

To be decided whether we are including #448

### CODE REVIEW

**Comment by jsha** [comment]
This looks good! Before we release: What do you think of making `gzip` enabled by default? I think it would be a really good thing. It would save people lots of bytes with no trouble. As a concrete example, Go's net/http does gzip by default and transparently, and it's quite wonderful.

I'll try to send a patch/proposal today. I'd also like to take another pass at the interface for enabling gzip/brotli so we can make them more independent of feature flags and diamond dependency issues.

**Comment by algesten** [comment]
Yeah let's enable gzip by default. Sounds good to me. I'll make a separate PR for that.

**Comment by algesten** [comment]
@jsha I assume you're ok with releasing 2.4.0 now?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,15 @@
+# 2.4.0
+
+ * Enable `gzip` feature by default (#455)
+ * `gzip` and `brotli` feature flags to enable decompression (#453, #421)
+ * Middleware function on agent (#448)
+ * Agent option to preserve `Authorization` header on redirects (#445)
+ * Deprecate re-exported serde types (#446)
+ * Widen type of `send_json` to `impl Serializable` (#446)
+ * `native-tls` feature provides an alterantive TLS backend (#449, #391)
```
Comment (jsha) [inline CHANGELOG.md:9]:
```suggestion
 * `native-tls` feature provides an alternative TLS backend (#449, #391)
```

**Review by jsha — APPROVED** [review]

**Pull request merged**